### PR TITLE
Enabling list attriubtes on the app clusters and updating the TV example app

### PR DESCRIPTION
**Pull request by lazarkov**
#### Problem
We do not have list attributes enabled on app clusters.

#### Summary of Changes
- Updated all the app clusters where list attributes are used
- Updated the TV example app to use the new attributes

#### Note
Everything which is not supported by the CHIP framework currently is noted with TODO comment.

#### Test
- Tested locally using chip-tool client and tv-server
- Used the `./scripts/tools/zap_regen_all.py ` and `./gn_build.sh ` to verify the building is successful

### CODE REVIEW

**Comment by lazarkov** [comment]
- Added the list attribute to all the xml files
- Updated the includes folder and every cluster manager now populates the list with example data
- Updated the cluster-init.cpp file, now every cluster once it is initialised has the logic which stores the example list

**Review by woody-apple — APPROVED** [review]

**Comment by lazarkov** [comment]
@woody-apple update the PR. Conflicts should be resolved now.

File: examples/chip-tool/commands/clusters/Commands.h [hunk 1]
```diff
@@ -662,6 +662,61 @@ static void OnTestClusterClusterTestSpecificResponse(void * context, uint8_t ret
     command->SetCommandExitStatus(true);
 }
 
+static void OnApplicationLauncherApplicationLauncherListListAttributeResponse(void * context, uint16_t count, uint16_t * entries)
+{
+    ChipLogProgress(chipTool, "OnApplicationLauncherApplicationLauncherListListAttributeResponse: %lu entries", count);
```
Comment (jelderton) [inline examples/chip-tool/commands/clusters/Commands.h:667]: nit: could use PRIu16

**Review by jelderton — APPROVED** [review]

**Comment by woody-apple** [comment]
@lazarkov Sorry, still failing build - and now conflicted again with tv app :(

**Comment by bzbarsky-apple** [comment]
@lazarkov I would recommend splitting this into two changesets within the PR: one with the manual changes and one with the auto-generated changes.  That will make review easier and also make merging/rebasing easier: the generated PR can just be dropped nad then regenerated after merging/rebasing the manual changes.

**Comment by lazarkov** [comment]
@bzbarsky-apple I understand it's hard to follow as gen/ folder polutes the PR. 
_I do not understand the advice of splitting of the PRs. For example if I add only xml changes without gen/ folder changes CI will fail as it compares if the gen/ folder the CI builds vs the one I've submitted. So it will expect to have the changes in gen/ folder_ 

Now regarding the PR, on how to follow what actually is the change:
**I've added support for all the list attributes in the clusters as they were not supported before.**
1. Follow the changes in the cluster .xml files. Every TODO: List is now removed and list attribute is added
2. I've created the ClusterManager.cpp file which stores the list attribute for any cluster
3. I've called the ClusterManager on cluster-init.cpp on every cluster initialization 
4. I've passed the "fake list data" for every cluster which is stored on cluster init.

Clusters who have now list attribute supported are: TV Channel, Content Launcher, Application Launcher, Audio Output, Media Input and Target navigator.

**Review by andy31415 — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> I do not understand the advice of splitting of the PRs.

@lazarkov I did not say to split up into multiple _PR_s.  I said to split up into multiple changesets within a _single_ PR.  This makes review significantly simpler, because instead of needing to find the non-generated files in the overall diff you just have two separate diffs: one with the manual changes, one with the generated ones.

See https://github.com/project-chip/connectedhomeip/pull/6804 for an example.  If you look at https://github.com/project-chip/connectedhomeip/pull/6804/commits there are two commits: one with the manual changes, and one with the generated stuff.
